.github/workflows/release.yml

```diff
@@ -0,0 +1,266 @@
+      - "v*.*.*"
+      - "nightly"
+
+env:
+  BUILD_TYPE: Release
+
+jobs:
+  test-prime-bsi:
+    name: "Test on Windows with Prime BSI"
+    runs-on:
+      - self-hosted
+      - pvcam
+      - windows # pwsh, msvc
+      - Prime-BSI # for tests
+
+    permissions:
+      actions: write
+
+    steps:
+      - name: Cancel Previous Runs
+        uses: styfle/cancel-workflow-action@0.10.0
+        with:
+          access_token: ${{ github.token }}
+
+      - name: Enable long paths
+        run: |
+          git config --global core.longpaths true
+        shell: pwsh
+
+      - name: Checkout
+        uses: actions/checkout@v3
+        with:
+          submodules: true
+          ref: ${{ github.event.pull_request.head.sha }}
+
+      - name: Get CMake 3.24
+        uses: lukka/get-cmake@latest
+        with:
+          useCloudCache: false
+          useLocalCache: true
+          cmakeVersion: 3.24.3
+
+      - name: Install MSVC
+        uses: ilammy/msvc-dev-cmd@v1
+        with:
+          arch: amd64_x86
+
+      - name: Configure
+        run: cmake -B ${{github.workspace}}/build -DCMAKE_BUILD_TYPE=${{env.BUILD_TYPE}}
+
+      - name: Build
+        run: cmake --build ${{github.workspace}}/build --config ${{env.BUILD_TYPE}}
+
+      - name: Test
+        working-directory: ${{github.workspace}}/build
+        run: ctest -C ${{env.BUILD_TYPE}} -L prime-bsi --output-on-failure
+
+  test-bsi-express:
+    name: "Test on Windows with BSI Express"
+    runs-on:
+      - self-hosted
+      - pvcam
+      - windows # pwsh, msvc
+      - BSI-Express # for tests
+
+    permissions:
+      actions: write
+
+    steps:
+      - name: Cancel Previous Runs
+        uses: styfle/cancel-workflow-action@0.10.0
+        with:
+          access_token: ${{ github.token }}
+
+      - name: Enable long paths
+        run: |
+          git config --global core.longpaths true
+        shell: pwsh
+
+      - name: Checkout
+        uses: actions/checkout@v3
+        with:
+          submodules: true
+          ref: ${{ github.event.pull_request.head.sha }}
+
+      - name: Get CMake 3.24
+        uses: lukka/get-cmake@latest
+        with:
+          useCloudCache: false
+          useLocalCache: true
+          cmakeVersion: 3.24.3
+
+      - name: Install MSVC
+        uses: ilammy/msvc-dev-cmd@v1
+        with:
+          arch: amd64_x86
+
+      - name: Configure
+        run: cmake -B ${{github.workspace}}/build -DCMAKE_BUILD_TYPE=${{env.BUILD_TYPE}}
+
+      - name: Build
+        run: cmake --build ${{github.workspace}}/build --config ${{env.BUILD_TYPE}}
+
+      - name: Test
+        working-directory: ${{github.workspace}}/build
+        run: ctest -C ${{env.BUILD_TYPE}} -L bsi-express --output-on-failure
+
+  windows-build:
+    name: "Build on Windows with PVCAM SDK"
+    runs-on:
+      - self-hosted
+      - pvcam
+      - windows
+    needs: # Don't bother building a release if tests fail
+      - test-prime-bsi
+      - test-bsi-express
```

Comment:
I separate out the Prime BSI and BSI Express tests and skip the Windows build if either fails. It's not using the exact artifact that the windows-build job is building, but it's the same code.